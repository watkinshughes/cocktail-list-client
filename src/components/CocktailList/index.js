import React, { Component } from "react";
import axios from "axios";
import Cocktail from "../Cocktail";
import styles from "./styles.module.css";

class CocktailList extends Component {
  state = {
    initialData: [],
    data: [],
    loading: true
  };

  sortData = data => {
    return data.sort(function(a, b) {
      if (a.name < b.name) {
        return -1;
      }
      if (a.name > b.name) {
        return 1;
      }
      return 0;
    });
  };

  fetchData = () => {
    const url = "https://cocktail-list-api.herokuapp.com/cocktails";
    axios.get(url).then(response => {
      this.setState({
        data: this.sortData(response.data),
        initialData: this.sortData(response.data),
        loading: false
      });
    });
  };

  buildCocktailsList = data => {
    return data.map(node => {
      return (
        <Cocktail key={node.name} info={node}>
          {node.children}
        </Cocktail>
      );
    });
  };

  filterList = event => {
    let updatedList = this.state.initialData;
    updatedList = updatedList.filter(
      node =>
        node.name.toLowerCase().search(event.target.value.toLowerCase()) !==
          -1 ||
        node.ingredients
          .toLowerCase()
          .search(event.target.value.toLowerCase()) !== -1
    );
    this.setState({
      data: updatedList
    });
  };

  componentDidMount() {
    this.fetchData();
    this.setState({
      data: this.state.initialData
    });
  }

  render() {
    return (
      <section className={styles.list}>
        <form>
          <fieldset>
            <label>
              <span className="visually-hidden">Filter by name or search by ingredient</span>
              <input
                type="text"
                className="Search"
                placeholder="Filter by name or search by ingredient"
                onChange={this.filterList}
              />
            </label>
          </fieldset>
        </form>
        {this.buildCocktailsList(this.state.data)}
      </section>
    );
  }
}

export default CocktailList;
