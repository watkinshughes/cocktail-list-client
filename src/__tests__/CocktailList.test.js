import React from "react";
import ReactDOM from "react-dom";
import CocktailList from "../components/CocktailList";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<CocktailList />, div);
  ReactDOM.unmountComponentAtNode(div);
});
