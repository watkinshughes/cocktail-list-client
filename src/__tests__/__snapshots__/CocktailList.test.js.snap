// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapsot 1`] = `
<section
  className="list"
>
  <form>
    <fieldset>
      <label>
        <span
          className="visually-hidden"
        >
          Filter by name or search by ingredient
        </span>
        <input
          className="Search"
          onChange={[Function]}
          placeholder="Filter by name or search by ingredient"
          type="text"
        />
      </label>
    </fieldset>
  </form>
</section>
`;
